### PR TITLE
Use Minio for S3 Storage integration tests

### PR DESCRIPTION
**Pull request by Jokser**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):
For better S3 integration testing Minio S3 Compatible object storage is used (https://min.io/).
This also future investment to have possibility to test various S3 features without necessary to S3 Mock support. 

Detailed description (optional):
Added docker-compose file that contains Minio and Nginx server (for redirect testing).
Added possibility to run an integration test using Minio in the separate docker containers. 
S3 Storage tests are reworked.
The self-written S3 HTTP Mock server is removed.

### CODE REVIEW

**Comment by Jokser** [comment]
To avoid `minio` module missing in integration tests image `yandex/clickhouse-integration-tests-runner` should be rebuild using modified Dockerfile in the current branch.

File: dbms/tests/integration/test_storage_s3/test.py [hunk 1]
```diff
@@ -1,159 +1,168 @@
...
-    assert 1 < data["multipart_parts"] < 10000
+
+
+# Test multipart put.
+def test_multipart_put(cluster):
+    # type: (ClickHouseCluster) -> None
+
+    instance = cluster.instances["dummy"]  # type: ClickHouseInstance
+    table_format = "column1 UInt32, column2 UInt32, column3 UInt32"
+    long_data = [[i, i + 1, i + 2] for i in range(100000)]
```
Comment (excitoon) [inline dbms/tests/integration/test_storage_s3/test.py]: We need to generate more than 5MB of data here and it should be done in more predictable way than just `100000`, see `get_random_string()`.

File: dbms/tests/integration/test_storage_s3/test.py [hunk 2]
```diff
@@ -1,159 +1,168 @@
...
+    long_values_csv = "".join(["{},{},{}\n".format(x, y, z) for x, y, z in long_data])
+    filename = "test.csv"
+    put_query = "insert into table function s3('http://{}:{}/{}/{}', 'CSV', '{}') format CSV".format(
+        cluster.minio_host, cluster.minio_port, cluster.minio_bucket, filename, table_format)
+
+    # Minimum size of part is 5 Mb for Minio.
+    # See: https://github.com/minio/minio/blob/master/docs/minio-limits.md
+    run_query(instance, put_query, stdin=long_values_csv, settings={'s3_min_upload_part_size': 5 * 1024 * 1024})
+
+    assert long_values_csv == get_s3_file_content(cluster, filename)
```
Comment (excitoon) [inline dbms/tests/integration/test_storage_s3/test.py]: Code does not check that ClickHouse do send more that one part of data

File: dbms/tests/integration/test_storage_s3/test.py [hunk 3]
```diff
@@ -1,159 +1,168 @@
...
+    # type: (ClickHouseInstance, str, object, dict) -> str
+
     logging.info("Running query '{}'...".format(query))
     result = instance.query(query, stdin=stdin, settings=settings)
     logging.info("Query finished")
-    return result
 
+    return result
 
-def test_get_with_redirect(started_cluster):
```
Comment (excitoon) [inline dbms/tests/integration/test_storage_s3/test.py]: Can you check that GET with redirect works?

Comment (Jokser) [reply dbms/tests/integration/test_storage_s3/test.py]: This is tested at `test_put_get_with_redirect`

File: dbms/tests/integration/test_storage_s3/test.py (hunk 2 above)
Comment (Jokser) [reply dbms/tests/integration/test_storage_s3/test.py]: Fixed.

File: dbms/tests/integration/test_storage_s3/test.py (hunk 1 above)
Comment (Jokser) [reply dbms/tests/integration/test_storage_s3/test.py]: Fixed.

**Pull request merged**